### DISCUSSION TITLE
Ctrlr VST Win11 support

### DISCUSSION THREAD

**Discussion by inteyes** [RomanKubiak/ctrlr#655] [Q&A] [answered]
Hey guys!

Can somebody tell me if Ctrlr 5.3.201 is working as a VST plugin under Windows 11? I cannot launch neither Ctrlr VST, nor my exported VST instances there.

Maybe I should install some additional package (other than VCredist) to make it work? Or it's not supposed to work there at all? Maybe it worth to mention that standalone version is working just fine.

Thanks!

**Comment by damiensellier** [comment, marked as answer]
5.3.201 is VST2 maybe your daw is not working with this format anymore?
What DAW are your using?

you can also try the latest CtrlrX version : 

CtrlrX v5.6.30.1 Windows Intel
[CtrlrX 5.6.30.1- EXE-Windows-x64.zip](https://github.com/RomanKubiak/ctrlr/files/14651826/CtrlrX.5.6.30.1-.EXE-Windows-x64.zip)
[CtrlrX-5.6.30.1-VST3-Windows-x64.zip](https://github.com/RomanKubiak/ctrlr/files/14651830/CtrlrX-5.6.30.1-VST3-Windows-x64.zip)
[CtrlrX-5.6.30.1-VST-Windows-x64.zip](https://github.com/RomanKubiak/ctrlr/files/14651832/CtrlrX-5.6.30.1-VST-Windows-x64.zip)

EDIT: The above builds are not working properly and need to be updated because source files were not the right one.

**CtrlrX 5.6.29 Windows 10**
[CtrlrX-x64-5.6.29.zip](https://github.com/RomanKubiak/ctrlr/files/14216062/CtrlrX-x64-5.6.29.zip)
[CtrlrX-x64.vst3-5.6.29.zip](https://github.com/RomanKubiak/ctrlr/files/14216131/CtrlrX-x64.vst3-5.6.29.zip)

**Reply by inteyes** [reply]
Thank you, @damiensellier!

This version can be loaded into the DAW (FL Studio 21) as a VST with no problems. Looks like my panel's code is not fully compatible with this version of Ctrlr though, as some errors appearing on startup.

But that's a great start poing, thank you once again for the help :)

**Reply by damiensellier** [reply]
usually the most common error with older panels is with panel:sendMidiMessageNow() in LUA scripts.

midiMessage must be sent as a memoryBlock not as a LUA table only : `panel:sendMidiMessageNow(CtrlrMidiMessage(MemoryBlock({0xB0, 0x0C, CcValue}))`

Don't hesitate if you need help to make it work.
Is it a panel you got on ctrlr.org ?

**Reply by inteyes** [reply]
Yeah, thanks :) 

I am trying to revive panels I made myself (which are presented on the ctrlr.org as well :) ). The [MS2000](https://github.com/inteyes/ReMS2000) panel just crash Ctrlr right after launch, while [X-Station](https://github.com/inteyes/X-Station-Voice-Editor) is not configuring properly on startup - as far as I can see after a little research, some global variables remain uninitialized.

I think I'll look into it more when I have some spare time.

PS: thanks for the tip on using MemoryBlock structure, it's well worth to know!

**Reply by damiensellier** [reply]
Oh okay those panels are from you! They look awesome really. I remember digging into the code of the X-Station and it's flawless.

When I open x station there's no alerts on my side.

I guess the problems on the ms2000 are from sendSysExMessage() the new JUCE probably requires the message to be embedded as a memoryBlock as well. my 2 cents...

I also noticed from upgrading that certain values needed to be converted with tonumber().

**Reply by damiensellier** [reply]
EDIT: The links above to WIN builds are not working properly and need to be updated because source files were not the right one.

**CtrlrX 5.6.29 Windows 10**
[CtrlrX-x64-5.6.29.zip](https://github.com/RomanKubiak/ctrlr/files/14216062/CtrlrX-x64-5.6.29.zip)
[CtrlrX-x64.vst3-5.6.29.zip](https://github.com/RomanKubiak/ctrlr/files/14216131/CtrlrX-x64.vst3-5.6.29.zip)
